cameo/strain_design/heuristic/evolutionary/objective_functions.py

```diff
@@ -116,18 +116,25 @@ def _repr_latex_(self):
 
 
 class YieldFunction(ObjectiveFunction):
-    def __init__(self, product, substrate, *args, **kwargs):
+    def __init__(self, product, substrates, *args, **kwargs):
         super(YieldFunction, self).__init__(*args, **kwargs)
-        self.n_c_substrate = 1
+
+        self.n_c_substrates = {}
         self.n_c_product = 1
         if isinstance(product, Reaction):
             self.n_c_product = product.reactants[0].elements.get('C', 1)
             product = product.id
         self.product = product
-        if isinstance(substrate, Reaction):
-            self.n_c_substrate = substrate.reactants[0].elements.get('C', 1)
-            substrate = substrate.id
-        self.substrate = substrate
+
+        if not isinstance(substrates, list):
```

Comment:
This should be changed to check for string or Reaction instead. This way other iterables than list will work